Zend/zend_ast.c

```diff
@@ -797,6 +805,88 @@ ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate(zval *result, zend_ast *ast
 			ZVAL_COPY_OR_DUP(result, zv);
 			break;
 		}
+		case ZEND_AST_NEW:
+		{
+			zend_class_entry *ce = zend_ast_fetch_class(ast->child[0], scope);
+			if (!ce) {
```

Comment:
nit: Would UNEXPECTED make sense for the `return FAILURE` branches added in this PR? I see them in the above case

Reply:
Personally I think we have too much UNEXPECTED in places where it's unlikely to actually matter...